Add etag support to balde

jaysonsantos opened the pull request:
The basic idea is to user call `balde_response_etag_matching` after it has created the body, and if there is a `If-None-Match` header it will try to match the body hash with sent header, if it is ok, it will truncate the body and change the status code to 304 Not Modified.
If user wants to add a etag header to it's response, it should call `balde_response_add_etag_header`.

As I'm a c developer, you can nitpick the code :)

rafaelmartins commented on src/scgi.c:
```diff
@@ -151,8 +152,8 @@ balde_scgi_parse_request(balde_app_t *app, GInputStream *istream)
-    GHashTable *headers = g_hash_table_new_full(g_str_hash, g_str_equal,
-        g_free, g_free);
+    GHashTable *headers = g_hash_table_new_full(g_str_hash,
+        balde_header_compare, g_free, g_free);
```
you need to change this in the other request parsers: `src/cgi.c` and `src/httpd.c`

jaysonsantos commented:
@rafaelmartins Can you check it again? I've changed `src/cgi.c` and `src/httpd.c` to use `balde_header_compare` and changed `balde_make_response_from_static_resource` to use `balde_response_etag_matching`.
